pkg/fileutil/purge_test.go

```diff
@@ -47,14 +51,28 @@ func TestPurgeFile(t *testing.T) {
 		}
 		time.Sleep(10 * time.Millisecond)
 	}
-	fnames, err := ReadDir(dir)
-	if err != nil {
-		t.Fatal(err)
+
+	// purge routine should purge 7 out of 10 files and only keep the
+	// 3 most recent ones.
+	// wait for purging for at most 100ms.
+	var fnames []string
+	for i := 0; i < 10; i++ {
+		fnames, err = ReadDir(dir)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(fnames) > 3 {
```

Comment:
I would invert this (`if len(fnames) <= 3 {break}`) to obviate the `continue`, but whatever